src/lib/RoutingManager.ts

```diff
@@ -44,14 +47,15 @@ class RoutingManager implements Widget {
     uiState: UiState;
   }): SearchParameters {
     const widgets = this.instantSearchInstance.mainIndex.getWidgets();
+    const indexUiState = uiState[this.indexId] || {};
 
     return widgets.reduce((parameters, widget) => {
       if (!widget.getWidgetSearchParameters) {
         return parameters;
       }
 
       return widget.getWidgetSearchParameters(parameters, {
-        uiState,
+        uiState: indexUiState,
```

Comment:
I didn't update the name of the argument but its type changed (`UiState` → `IndexUiState`). We can change it (BC) but IMO it's not worth it. We already have the type to warn us about its usage. We can update it if you think that it's too confusing.

Reply:
We can also change the route state name to something else, keeping UiState for the focused one, and "GlobalUiState" for global issues

Reply:
The needs to put "global" in the name, feels odd to me.

Reply:
I think this "mis-naming" is fine for now. We can still add it to the list of things to think about/check before releasing v4 stable.